### DISCUSSION TITLE
An option to disable category from being selected in article edit mode

### DISCUSSION THREAD

**Discussion by Sulpher** [joomla/joomla-cms#28371] [2. Cold Features]
### Describe the solution you'd like

The category structure of a news portal is organized by the classical principle:

News
    - Politics
    - Sport
    - Society
and etc.
When a user adds a new article, there is a need to select the correct subcategory.
However, the user can do a mistake by saving the article without choosing the right category and the article is being assigned to the first category from the list (News), or there could be a situation when the user selects the root category by mistake during working on a lot of articles.

My suggestion is to add an extra option in category settings allowing to disable the category from being selected in article edit mode.
![img2](https://user-images.githubusercontent.com/3432048/76755866-b1dc1500-6795-11ea-920d-78a6fc1d0796.png)
![img1](https://user-images.githubusercontent.com/3432048/76755875-b6a0c900-6795-11ea-91c4-9c7e2c530685.png)

If it is not possible to disable the category, it could be good to ink it in a grey color which could be a signal to the webmaster not to select this category.

**Comment by GCLW** [comment]
I don't know how you have things setup, but you should be able to do this with Groups and Permissions to specify what users have access to.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28371">issues.joomla.org/tracker/joomla-cms/28371</a>.</sub>

**Comment by Sulpher** [comment]
> I don't know how you have things setup, but you should be able to do this with Groups and Permissions to specify what users have access to.

The root category override permissions for subcategory.
CATEGORY:            PERMISSIONS:
====
News                       Create = not allowed
     - Economy         Create = allowed (takes no effect)

![Screen Shot 2020-03-16 at 17 17 47](https://user-images.githubusercontent.com/3432048/76767169-3e90ce00-67aa-11ea-8f4e-8008f16a1644.png)
![Screen Shot 2020-03-16 at 17 12 58](https://user-images.githubusercontent.com/3432048/76767184-42245500-67aa-11ea-8c55-f603af785eed.png)

Since I set Create param to Denied for the root category, the child category gets an override of the settings despite the fact I've set the param to allowed.

**Comment by GCLW** [comment]
I spoke too soon.
I also have a news based site and my news is broken down into:

News
- Local News
- Sports
- Weather

I have setup groups so reporters "users" only have access to one of those sub categories. Basically so a sports reporter only has "sports" available in the category drop-down menu.

Looking at your above issue, I think I ran into the exact same thing due to the parent setting controlling the children.

Looking at the solution I did come up with, I just placed the news sub categories in the root, and removed the news category. This setup worked for me, but I can see not everyone can use that solution. I am 100% positive I did that because I did not want the reporters to run into the issue you are reporting.

**Comment by Sulpher** [comment]
@GCLW, thank you for your reply. However, I have no note that child categories are useful to "markup" the structure where you see related categories grouped by the root names.

In other words, it is important to keep the nested structure for quick navigation when you're working with the content hard and publish a lot of articles step-by-step, but your site structure has not only news section, but other categories too.

Example:

News
- Local News
- Sports
- Weather

Travel

- Visa/IDs
- Get started
- Accommodation

Blog

- Personal blog
- Useful articles
- Project announces

...

Etc.

Just imagine if we set up all these nested categories as root. The result is a lack of usability since a publisher should spend seconds by searching for the right category, but when it is grouped in categories/sub-categories, you save a lot of time. 
However, as I mentioned above, the human factor happens and you can forget to select the right category during the work and the default value will be set. 
So, that's why I suggested the new option.
However, thank you for your idea to make a category called Draft as the first one.

**Comment by KarstenFrostholm** [comment]
Isn't that something you can do with a plugin? with a filter based on user or groups!

**Comment by AndySDH** [comment]
This is a nice idea!
